### ISSUE TITLE
Version numbers of packages in mirage-dev OPAM remote

### ISSUE THREAD

**Issue by mato** [mirage/mirage-dev#128]
I'd like to bulk-change the version numbers of all packages in this repository to `mirage-dev`. This has a couple of advantages:

1. No maintenance required when new versions of packages are released to `opam.ocaml.org` as the version `mirage-dev` will always be greater than a version of `[0-9]*`.
2. During development I can immediately see which development packages I have installed with `opam list | grep mirage-dev`.

@samoht Can you confirm that this is the way the OPAM version comparison works?

**Comment by avsm** [comment]
I think this works...

**Comment by avsm** [comment]
You could also use `dev~mirage` so that it's less than `dev` (which is a common one for a local pin), but that's possibly too exotic

**Comment by samoht** [comment]
Seems to be ok:

```
$ ledit ocaml
# #require "opam-lib";;
# OpamPackage.Version.(compare (of_string "mirage-dev") (of_string "9.9"));;
1
```

**Comment by djs55** [comment]
I like this -- much clearer than using an arbitrary number (which is what I've been doing recently)

**Comment by mato** [comment]
@avsm If you mean to ensure the use of `dev` as a version for local pins (added with `opam pin`), don't those already have absolute priority over anything coming from a remote?

**Comment by avsm** [comment]
Yeah pins do, but might as well make the ordering really obvious in the versions from an `opam list` as well.  I don't think it's a big deal at this stage.

**Comment by mato** [comment]
@avsm I'll go with `dev~mirage` when doing the changes, saves us changing it again.

Going through the packages currently in mirage-dev, some are pinned to specific versions (either via git tags or archive URLs) while others are pointing to git master / other branches:

1. Pinned to specific versions: `bin_prot`, `camlzip`, `zlib-xen`.
2. Git-pinned to non-master branch: `dolog` (_UnixJunkie/dolog.git#no_unix_), `dns` (_mirage/ocaml-dns#mirage-3.0_), `nocrypto` (_hannesm/ocaml-nocrypto#p_).
3. Git-pinned to master branch: everything else

@samoht @hannesm Can you confirm that out of the above, 1) should keep their existing (specific) version numbers, 2) and 3) can be changed to `dev~mirage`?

**Comment by hannesm** [comment]
`bin_prot`/`camlzip`/`zlib-xen`/`dolog` were needed for irmin/git on Xen, not sure what the current state is (@samoht should know whether those here are bitrot or not; I do not use mirage-dev when building Canopy for xen which uses irmin/git).

the rest is targeting mirage-3, where `dns` and `nocrypto` target custom branches, the rest master.

please go ahead and set the version of those (2 & 3 above) to `dev-mirage`.

**Comment by mato** [comment]
Fixed via #129. Note that Travis is currently failing for mirage-dev master and has been for some time (unrelated to these changes).
